packages/queued-request-controller/src/QueuedRequestController.ts

```diff
@@ -0,0 +1,136 @@
+import type { RestrictedControllerMessenger } from '@metamask/base-controller';
+import { BaseControllerV2 } from '@metamask/base-controller';
+import type { Patch } from 'immer';
+
+const controllerName = 'QueuedRequestController';
+
+// should serialize and persist these values.
+const stateMetadata = {
+  queue: { persist: false, anonymous: false },
+};
+
+const getDefaultState = () => ({});
```

Comment:
[nit] necessary?

Reply:
Not necessary, since  there  is  currently no actual state (at least for now. In the future we may way to look toward serializing the requests and saving them)